src/user-statuses/UserStatusScreen.js

```diff
@@ -36,74 +58,136 @@ type Props = $ReadOnly<{|
+    setEmojiInputValue(inputValueFromStatusEmoji(null));
+  }, []);
 
   return (
     <Screen title="User status">
-      <Input
-        autoFocus
-        maxLength={60}
-        style={styles.statusTextInput}
-        placeholder="What’s your status?"
-        value={statusText || ''}
-        onChangeText={setStatusText}
-      />
-      <FlatList
-        data={statusSuggestions}
-        keyboardShouldPersistTaps="always"
-        keyExtractor={item => item}
-        renderItem={({ item, index }) => (
-          <SelectableOptionRow
-            key={item}
-            itemKey={item}
-            title={item}
-            selected={item === statusText}
-            onRequestSelectionChange={itemKey => {
-              setStatusText(_(itemKey));
-            }}
+      <View style={styles.inputRow}>
+        {zulipFeatureLevel >= 86 && (
+          <EmojiInput
+            navigation={navigation}
+            value={emojiInputValue}
+            onChangeValue={setEmojiInputValue}
+            rightMargin
           />
         )}
-      />
-      <View style={styles.buttonsWrapper}>
-        <ZulipButton
-          style={styles.button}
-          secondary
-          text="Clear"
-          onPress={handlePressClear}
-          Icon={IconCancel}
-        />
-        <ZulipButton
-          style={styles.button}
-          text="Update"
-          onPress={handlePressUpdate}
-          Icon={IconDone}
+        <Input
+          autoFocus
+          maxLength={60}
+          style={styles.statusTextInput}
+          placeholder="What’s your status?"
+          value={textInputValue}
+          onChangeText={setTextInputValue}
         />
+        {(emojiInputValue !== null || textInputValue.length > 0) && (
+          <Pressable style={styles.clearButton} onPress={handlePressClear} hitSlop={12}>
+            {({ pressed }) => (
+              <Icon name="x" size={24} color={pressed ? HIGHLIGHT_COLOR : BRAND_COLOR} />
+            )}
+          </Pressable>
+        )}
       </View>
+      <FlatList
+        data={statusSuggestions}
+        keyboardShouldPersistTaps="always"
+        keyExtractor={(item, index) => index.toString() /* list is constant; index OK */}
+        renderItem={({ item: [emoji, text], index }) => {
+          const translatedText = _(text);
+          return (
+            <SelectableOptionRow
+              itemKey={index}
+              title={
+                zulipFeatureLevel >= 86
+                  ? `${parseUnicodeEmojiCode(emoji.emoji_code)} ${translatedText}`
+                  : text
+              }
+              selected={
+                translatedText === statusTextFromInputValue(textInputValue)
+                && isEqual(emoji, statusEmojiFromInputValue(emojiInputValue))
+              }
+              onRequestSelectionChange={() => {
+                setTextInputValue(translatedText);
+                setEmojiInputValue(inputValueFromStatusEmoji(emoji));
```

Comment:
If emoji statuses aren't supported…we do end up just cutting the
    emoji out of the status suggestion, which I guess is a small
    downside. For these older servers, we could have kept up what we do
    now: stick the status emoji at the start of the suggested text. But
    I think achieving that has a higher complexity cost than it's worth.
    I've never really used this UI, I'm not sure how common it is to do
    so, and people are still free to depart from the suggested text,
    including by adding emojis, by using the text input.

Yeah, agreed.